documentation/processing.tex

```diff
@@ -26,16 +22,60 @@ \subsection{Overview}
 	\item \emph{Processing tracer} for tracing processed statements (see \cref{macro_tracer}).
 \end{itemize}
 
-\subsectionSrc{Types of statement processors}{parser\_library/src/processing/statement\_processors/statement\_processor.h}
-\label{lab06:sect_proc}
 
+\subsection{Composition}
+
+As the processing of the HLASM source file is rather complicated, we define a complex set of abstraction objects over the Gordian assembling of HLASM language:
+
+\begin{description}
+	\item[Statement provider] (see \cref{lab06:sect_prov}) is able to produce \TT{statement} structures. Its functionality is to provide statements from its finite statement stream.
+	
+	\item[Statement processor] (see \cref{lab06:sect_proc}) is an object that takes statement structures from a provider. Then, it performs a specific action with the acquired statement; namely, stores it into macro/copy definition (\emph{Macro/Copy processor}) or looks for sequence symbol (\emph{Lookahed processor}) or performs contained instruction (\emph{Opencode processor}).
+	
+	\item[Instruction processors] (see \cref{lab06:instr_proc}) help opencode statement processor in performing actions with the instructions contained in a statement. Each one of four instruction processors (Macro, Assembler, Machine and Conditional Assembly IP) process separate sub-set of a broad set of HLASM instructions. 
```

Comment:
process -> processes

Napisat rozdiel medzi statement processorom a instruction processorom? Nieco take ze statement processor sa diva na kontext v ktorom sa spracuva statement, nezaujima ho konkretna instrukcia (?), tak IP uz spracovava konkr. instrukcie? len napad neviem ci je to pravda :D

Reply:
ja si myslim, ze rozdiel co si napisal je z toho jasny
v SP je
 it performs a specific action with the acquired statement
v IP je
performing actions with the instructions contained in a statement